generic_config_updater/patch_sorter.py

```diff
@@ -333,6 +333,112 @@ def _extend_moves(self, move, diff):
             for newmove in extender.extend(move, diff):
                 yield newmove
 
+class ConfigFilter:
+    """
+    A filtering class to get the paths matching the filter from the given config.
+    The patterns:
+    - Each pattern consist of multiple tokens
+    - Tokens are matched with the config from the root level
+    - Each token can be:
+      - '*' Will match all keys at the current level
+      - '@' Will be replaced by 'common_key' passed in 'get_paths'
+      - <other> Will match other strings that are not * nor @
+    - Token advanced options
+      - '*|@' Will match keys that end with '|common_key'
+      - '@|*' Will match keys that start with 'common_key|'
+      - '*|<other>' Will match keys that end with '|<other>'
+      - '<other>|*' Will match keys that start with '<other>|'
+    """
+    def __init__(self, patterns):
+        self.patterns = patterns
+
+    def get_paths(self, config, common_key=None):
+        for pattern in self.patterns:
+            for path in self._get_paths_recursive(config, pattern, [], 0, common_key):
+                yield path
+
+    def _get_paths_recursive(self, config, pattern_tokens, matching_tokens, idx, common_key):
+        if idx == len(pattern_tokens):
+            yield PathAddressing().create_path(matching_tokens)
+            return
+
+        token = pattern_tokens[idx]
+        if common_key:
+            token = token.replace("@", common_key)
+
+        matching_keys = []
+        if token == "*":
+            matching_keys = config.keys()
+        elif token.startswith("*|"):
+            suffix = token[2:]
+            matching_keys = [key for key in config.keys() if key.endswith(suffix)]
+        elif token.endswith("|*"):
+            prefix = token[:-2]
+            matching_keys = [key for key in config.keys() if key.startswith(prefix)]
+        elif token in config:
+            matching_keys = [token]
+
+        for key in matching_keys:
+            matching_tokens.append(key)
+            for path in self._get_paths_recursive(config[key], pattern_tokens, matching_tokens, idx+1, common_key):
+                yield path
+            matching_tokens.pop()
+
+class PortCriticalConfigIdentifier:
+    def __init__(self, path_addressing):
+        # TODO: port-critical fields are hard-coded for now, it should be moved to YANG models
+        # NOTE: PortCritical validation logic can be generalized to be validation for any config that
+        #       requires another config to be of a specific value. Call the validator "TransitionalValueValidator"
+        #       For the case of port-critical validation the transitional value would be "admin_status=down".
+        self.port_critical_filter = ConfigFilter([
```

Comment:
@renukamanavalan let me know if there are more configs that needs to be marked with `port-critical`.

Reply:
Check with Neetha for complete set.

Reply:
We should also add PORT_QOS_MAP,
BUFFER_PORT_INGRESS_PROFILE_LIST, BUFFER_PORT_EGRESS_PROFILE_LIST [these 2 tables are not available on all asics]

Reply:
added